### PR TITLE
MAILET-145: Improve readability of tests involving mime messages

### PR DESCRIPTION
**Pull request by quynhn**
Take comment from: https://github.com/linagora/james-project/pull/537

### CODE REVIEW

File: mailet/standard/src/test/java/org/apache/james/transport/mailets/StripAttachmentTest.java [hunk 1]
```diff
@@ -61,11 +59,12 @@
...
-    private static final String MIME_HEADER_DEFAULT = "Content-Transfer-Encoding: 8bit\r\nContent-Type: application/octet-stream; charset=utf-8\r\n\r\n";
-    private static final String MIME_HEADER_TEXT_CALENDAR = "Content-Transfer-Encoding: 8bit\r\nContent-Type: text/calendar; charset=utf-8\r\n\r\n";
-    private static final String MIME_HEADER_TEXT_HTML = "Content-Transfer-Encoding: 8bit\r\nContent-Type: text/html; charset=utf-8\r\n\r\n";
-
+    private static final Optional<String> ABSENT_MIME_TYPE = Optional.absent();
+    public static final String CONTENT_TRANSFER_ENCODING = "Content-Transfer-Encoding";
+    public static final String CONTENT_TYPE = "Content-Type";
+    public static final String CONTENT_TYPE_DEFAULT = "application/octet-stream; charset=utf-8";
+    private static final String CONTENT_TRANSFER_ENCODING_VALUE ="8bit";
+    public static final String TEXT_CALENDAR_CHARSET_UTF_8 = "text/calendar; charset=utf-8";public static final String TEXT_HTML_CHARSET_UTF_8 = "text/html; charset=utf-8";
```
Comment (rouazana) [inline mailet/standard/src/test/java/org/apache/james/transport/mailets/StripAttachmentTest.java]: missing carriage return

File: mailet/standard/src/test/java/org/apache/james/transport/mailets/StripAttachmentTest.java [hunk 2]
```diff
@@ -110,26 +105,27 @@ public void serviceShouldNotModifyMailWhenNotMultipart() throws MessagingExcepti
...
-        message.setContent(multiPart);
-        message.saveChanges();
+        MimeMessage message = MimeMessageBuilder.mimeMessageBuilder()
+            .setMultipartWithBodyParts(
+                MimeMessageBuilder.bodyPartBuilder()
+                    .data("simple text")
+                    .build(),
+                createAttachmentBodyPart(expectedAttachmentContent, "10.tmp",
+                    new MimeMessageBuilder.Header(CONTENT_TRANSFER_ENCODING, CONTENT_TRANSFER_ENCODING_VALUE),
+                    new MimeMessageBuilder.Header(CONTENT_TYPE, CONTENT_TYPE_DEFAULT)),
```
Comment (rouazana) [inline mailet/standard/src/test/java/org/apache/james/transport/mailets/StripAttachmentTest.java]: you could have static methods for these classical headers

File: mailet/standard/src/main/java/org/apache/james/transport/matchers/GenericRegexMatcher.java [hunk 3]
```diff
@@ -27,10 +27,14 @@
 
 import javax.mail.MessagingException;
 import javax.mail.internet.MimeMessage;
+
+import java.util.Arrays;
```
Comment (rouazana) [inline mailet/standard/src/main/java/org/apache/james/transport/matchers/GenericRegexMatcher.java]: not needed

File: mailet/standard/src/main/java/org/apache/james/transport/matchers/GenericRegexMatcher.java [hunk 4]
```diff
@@ -27,10 +27,14 @@
...
 import java.util.Collection;
 import java.util.regex.Pattern;
 import java.util.regex.PatternSyntaxException;
 
+import com.google.common.collect.ImmutableList;
+
```
Comment (rouazana) [inline mailet/standard/src/main/java/org/apache/james/transport/matchers/GenericRegexMatcher.java]: not needed

File: mailet/standard/src/test/java/org/apache/james/transport/mailets/StripAttachmentTest.java (hunk 1 above)
Comment (quynhn) [inline mailet/standard/src/test/java/org/apache/james/transport/mailets/StripAttachmentTest.java]: Done

File: mailet/standard/src/main/java/org/apache/james/transport/matchers/GenericRegexMatcher.java (hunk 3 above)
Comment (quynhn) [inline mailet/standard/src/main/java/org/apache/james/transport/matchers/GenericRegexMatcher.java]: It's removed

File: mailet/standard/src/main/java/org/apache/james/transport/matchers/GenericRegexMatcher.java (hunk 4 above)
Comment (quynhn) [inline mailet/standard/src/main/java/org/apache/james/transport/matchers/GenericRegexMatcher.java]: It's removed

File: mailet/standard/src/test/java/org/apache/james/transport/mailets/StripAttachmentTest.java (hunk 2 above)
Comment (quynhn) [inline mailet/standard/src/test/java/org/apache/james/transport/mailets/StripAttachmentTest.java]: Done

**Comment by chibenwa** [comment]
test this please

**Comment by quynhn** [comment]
test this please

**Comment by quynhn** [comment]
test this please

**Comment by quynhn** [comment]
test this please

**Pull request merged**